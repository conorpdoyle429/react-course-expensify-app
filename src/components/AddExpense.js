import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';

export class AddExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push('/');
    };   
    render() {
        return (
            <div>
                <h1>This is from my add expense component</h1>
                <ExpenseForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        )
        
    }
}

// USED A CLASS HERE INSTEAD OF FUNCTION 
// don't want to have onSubmit created inline as it will cause unecessary rerendering

// const AddExpense = (props) => (
//     <div>
//         <h1>This is from my add expense component</h1>
//         <ExpenseForm 
//             onSubmit={(expense) => {
//                 console.log(expense);
//                 // props.dispatch(addExpense(expense));
//                 props.onSubmit(expense);
//                 props.history.push('/');
//             }}
//         />
//     </div>
// );


// pass this as second argument in connect
// this allows us to be able to test props.dispatch(...) with jest
const mapDispatchToProps = (dispatch) => ({
        addExpense: (expense) => dispatch(addExpense(expense))
    });

export default connect(undefined, mapDispatchToProps)(AddExpense);