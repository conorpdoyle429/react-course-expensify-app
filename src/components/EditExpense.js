import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpense extends React.Component {
    onClick = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/dashboard');
    }
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/dashboard');
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>  
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button 
                        className="button button--secondary"
                        onClick={this.onClick}
                    >
                        Remove Expense
                    </button>
                </div>     
            </div>
        )
    };
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};

const mapDispatchToProps = (dispatch) => ({
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);