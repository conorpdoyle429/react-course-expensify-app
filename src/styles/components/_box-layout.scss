.box-layout {
    align-items: center; // center vertically
    background: url('/images/bg.jpg');
    background-size: cover; // always cover container with image
    display: flex; // allows align-items and justify-content to work 
    height: 100vh; // 100% of viewport height
    justify-content: center; // center horizontally
    width: 100vw; // 100$ of viewport width
}

.box-layout__box {
    background: fade-out($color: white, $amount: .15); // fade out 15%
    border-radius: 3px; // round corners
    padding: $l-size $m-size;
    text-align: center; // text in center of box
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1; // all text has line height 
}