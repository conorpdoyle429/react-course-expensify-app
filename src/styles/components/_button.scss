.button{
    background: $blue;
    border: none;
    color: white;
    display: inline-block; // make sure vertical margin and padding works
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none; // remove underline
}

// button modifier to override background - for logout button
.button--link {
    background: none;
}

.button--secondary {
    background: $grey;
}