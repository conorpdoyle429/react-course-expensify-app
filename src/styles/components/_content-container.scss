.content-container {
    margin: 0 auto; // evenly distribute space on left and right
    max-width: 80rem;
    padding: 0 $m-size // padding on left and right of browser
}