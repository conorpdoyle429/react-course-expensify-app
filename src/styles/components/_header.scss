.header {
    background: $dark-blue;
}

.header__content {
    align-items: center; //center vertically 
    display: flex;
    justify-content: space-between; // distribute space between title and logout
    padding: $s-size 0; // top and bottom padding
}

.header__title {
    color: white;
    text-decoration: none; // clear the underline
    h1 {
        margin: 0;
    }
}