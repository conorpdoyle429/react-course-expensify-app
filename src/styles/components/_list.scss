.list-header {
    background: $light-grey;
    border: 1px solid darken($color: $light-grey, $amount: 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size; // smaller padding on top and bottom
}

// add a margin between content and bottom of page
.list-body {
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}

.list-item {
    border: 1px solid darken($color: $light-grey, $amount: 7%);
    border-top: none; // as we have list items stacked we don't want top borders
    color: $dark-grey;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none; // remove underline
    transition: background .3s ease; // ease in background in .3 seconds for hover
    &:hover {
        background: $light-grey; // change background when hovering 
    }
    // split apart amount and titles for large screens
    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
    }
}

// for no expenses message
.list-item--message {
    align-items: center;
    color: $grey;
    justify-content: center;
    padding: $m-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    font-size: $font-size-large;
    margin: 0;
    word-break: break-all; // break up really long words
}

.list-item__sub-title {
    color: $grey;
    font-size: $font-size-small;
}

.list-item__data {
    margin: $s-size 0 0 0 0; // only top margin
    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $s-size;
    }
}