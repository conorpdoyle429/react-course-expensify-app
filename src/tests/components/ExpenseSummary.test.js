import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

// react-test-renderer 

test('should render ExpenseSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={23456} />);
    expect(wrapper).toMatchSnapshot();
});