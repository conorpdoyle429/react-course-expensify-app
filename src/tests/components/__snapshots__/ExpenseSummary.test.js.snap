// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totaling 
    $234.56
  </h1>
</div>
`;
